# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Compacting.cpp
# Commit: 2fa53cd9ac4e
# Full Hash: 2fa53cd9ac4e821763dbc4402e58b58184ce230b
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-04 04:18:28
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 6: Stop using JSContext for GC helper thread tasks r=jandem
#   
#   This patch moves the debug-only GC state from JSContext to JSFreeOp. Support
#   functions for this move to FreeOp.h and most RAII classes move to GCInternals.h.
#   
# ==============================================================================

diff -r 054a08d13999 -r 2fa53cd9ac4e js/src/gc/Compacting.cpp
--- a/js/src/gc/Compacting.cpp	Thu Mar 03 17:49:18 2022 +0000
+++ b/js/src/gc/Compacting.cpp	Thu Mar 03 17:49:18 2022 +0000
@@ -210,7 +210,7 @@
 
 static void RelocateCell(Zone* zone, TenuredCell* src, AllocKind thingKind,
                          size_t thingSize) {
-  JS::AutoSuppressGCAnalysis nogc(TlsContext.get());
+  JS::AutoSuppressGCAnalysis nogc;
 
   // Allocate a new cell.
   MOZ_ASSERT(zone == src->zone());